csa-runner-app/src/main/java/com/farao_community/farao/swe_csa/app/dichotomy/index/SweCsaHalfRangeDivisionIndexStrategy.java

```diff
@@ -0,0 +1,144 @@
+package com.farao_community.farao.swe_csa.app.dichotomy.index;
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import com.powsybl.openrao.commons.Unit;
+import com.powsybl.openrao.data.cracapi.Crac;
+import com.powsybl.openrao.data.cracapi.cnec.AngleCnec;
+import com.powsybl.openrao.data.cracapi.cnec.FlowCnec;
+import com.farao_community.farao.dichotomy.api.results.DichotomyStepResult;
+import com.farao_community.farao.rao_runner.api.resource.RaoResponse;
+import com.farao_community.farao.swe_csa.app.dichotomy.CounterTradingDirection;
+import com.farao_community.farao.swe_csa.app.dichotomy.variable.MultipleDichotomyVariables;
+import com.farao_community.farao.swe_csa.app.rao_result.RaoResultWithCounterTradeRangeActions;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Network;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of IndexStrategy that consists of a basic dichotomy between minimum index value and maximum one, on
+ * FR-ES and PT-ES borders.
+ * When one of the borders is secured with a good precision, dichotomy continues on the other border until that one
+ * is secured too.
+ *
+ * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
+ * @author Jean-Pierre Arnould {@literal <jean-pierre.arnould at rte-france.com>}
+ */
+
+public class SweCsaHalfRangeDivisionIndexStrategy extends HalfRangeDivisionIndexStrategy<MultipleDichotomyVariables> {
+    private Set<FlowCnec> frEsFlowCnecs;
+    private Set<FlowCnec> ptEsFlowCnecs;
+    private Set<AngleCnec> frEsAngleCnecs;
+    private Set<AngleCnec> ptEsAngleCnecs;
+
+    public SweCsaHalfRangeDivisionIndexStrategy(Crac crac, Network network) {
+        super(false);
+        getCnecsBorder(crac, network, Country.FR);
+        getCnecsBorder(crac, network, Country.PT);
+    }
+
+    private void getCnecsBorder(Crac crac, Network network, Country country) {
+        //TODO : implement association between cnecs and borders (CSA-67)
```

Comment:
For now, it reports only cnecs (at least with one side) in Portugal for the border PT-ES, and (at least with one side) in France for the border FR-ES. But CSA-67 will change that because some Spanish cnecs could be involved with one border.